akka-typed/src/main/scala/akka/typed/cluster/sharding/ClusterSharding.scala

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.actor.ExtendedActorSystem
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardCoordinator.{ LeastShardAllocationStrategy, ShardAllocationStrategy }
+import akka.cluster.sharding.{ ClusterSharding => UntypedClusterSharding }
+import akka.cluster.sharding.{ ShardRegion => UntypedShardRegion }
+import akka.typed.internal.adapter.{ ActorRefAdapter, ActorSystemAdapter }
+import akka.typed.scaladsl.adapter.PropsAdapter
+import akka.typed.{ ActorRef, ActorSystem, Behavior, Extension, ExtensionId, Props }
+
+sealed case class ShardingEnvelope[A](entityId: String, message: A)
+
+/** Allows starting a specific Sharded Entity by its entity identifier */
+object StartEntity {
+
+  /**
+   * Scala API
+   *
+   * Returns [[ShardingEnvelope]] which can be sent via Cluster Sharding in order to wake up the
+   * specified (by `entityId`) Sharded Entity, ''without'' delivering a real message to it.
+   */
+  def apply[A](entityId: String): ShardingEnvelope[A] =
+    new ShardingEnvelope[A](entityId, null.asInstanceOf[A])
```

Comment:
shouldn't the wrapped message be `akka.cluster.sharding.ShardRegion.StartEntity`, or is this for something else?

also wonder if those factory methods should instead be in `ShardingEnvelope` companion?

Reply:
Relates to above point - such special things then mean that we'd keep ShardingEnvelope sealed and provide special subclasses... I think that makes sense, so will introduce a real StartEntity type then

Reply:
I don't know if that will work. In the end it must be untyped sharding that should understand this special message. Perhaps @johanandren had some thoughts about this?
It's difficult since it's of wrong type. Perhaps this feature should not be included in the first implementation, and we create a follow up ticket about it?

Reply:
Sounds good to make it a ticket and follow up from my side; WDYT @johanandren

Reply:
I thought I documented my thinking about (the hack) using null to signify start of entity, which could work with both the envelope and when not using envelopes (protocol specific message to start but no message actually delivered)

Reply:
ok, might work when sent from the outside, but we must also support the `akka.cluster.sharding.ShardRegion.StartEntity` in our provided extractors, since that can be sent by Cluster Sharding itself. Create a ticket if it's not obvious how to impl.

Reply:
Yeah, I didn't get *that* far in thinking about it. Ticketified in #23679